src/conversions/cart.cpp

```diff
@@ -121,8 +121,15 @@ static double geocentric_radius(double a, double b, double cosphi,
         This is from WP2, but uses hypot() for potentially better
         numerical robustness
     ***********************************************************************/
+#ifdef not_optimized
     return hypot(a * a * cosphi, b * b * sinphi) /
            hypot(a * cosphi, b * sinphi);
+#else
+    return a *
+           sqrt(cosphi * cosphi +
+                (b_div_a * b_div_a) * (b_div_a * b_div_a) * (sinphi * sinphi)) /
```

Comment:
Those parenthesis are not needed. Are for readability or helping the compiler?

Reply:
That might help the compiler. Otherwise if we do "a * a * a * a", I suspect he's supposed to evaluate it as "((a * a) * a) * a" given the left-to-right evaluation order mandated by C/C++. Grouping by pair will save one multiplication (since the optimizer should be smart enough to evaluate once b_div_a * b_div_a

Reply:
actually I've just added a commit to further optimize things